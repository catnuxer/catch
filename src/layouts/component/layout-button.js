/** @jsx jsx */
import React from "react";
import { css, jsx, keyframes } from "@emotion/core";
import { desktopLarge, desktop, ipad, phone } from "../style/style-query";
import colors from "../style/style-colors";

const Btn = ({ label, onPress }) => {
  return (
    <button
      css={css`
        width: 100%;
        padding: 15px;
        border: 1px solid ${colors.grey};
        border-radius: 5px;
        &:hover {
          cursor: pointer;
        }
        &:active {
          border: 1px solid ${colors.blue};
        }
      `}
      onClick={onPress}
    >
      {label}
    </button>
  );
};

const propsButton = (prevProps, nextProps) => {
  return prevProps.label === nextProps.label;
};

export const Button = React.memo(Btn, propsButton);

const List = ({ label, onPress }) => {
  return (
    <li
      css={css`
        font-size: 14px;
        margin: 0px;
        padding: 15px;
        &:hover {
          background-color: ${colors.blueSoft};
          cursor: pointer;
        }
      `}
      onClick={onPress}
    >
      <span>{label}</span>
    </li>
  );
};

const propsList = (prevProps, nextProps) => {
  return prevProps.label === nextProps.label;
};

export const ListButton = React.memo(List, propsList);

export const ListPopUp = React.forwardRef((props, ref) => {
  return (
    <div
      ref={ref}
      css={css`
        position: absolute;
        background: white;
        width: 100%;
        top: 100px;
        left: 0;
        z-index: 2;
        box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 6px 0px;
      `}
    >
      <ul
        css={css`
          list-style-type: none;
          margin: 0px;
          padding: 0px;
        `}
      >
        {props.data.map((item, index) => (
          <ListButton
            key={index.toString()}
            onPress={item.onPress}
            label={item.label}
          />
        ))}
      </ul>
    </div>
  );
});
