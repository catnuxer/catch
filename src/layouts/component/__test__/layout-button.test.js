import React from "react";
import serializer from "jest-emotion";
import renderer from "react-test-renderer";
import { Button, ListPopUp, ListButton } from "../layout-button";

expect.addSnapshotSerializer(serializer);

describe("layout-button test", () => {
  it("snappshoot button", () => {
    expect(
      renderer.create(<Button label="test" onPress={jest.fn()} />).toJSON()
    ).toMatchSnapshot();
  });
});

describe("layout-button ListPopUp test", () => {
  it("snapshoot listpopup", () => {
    const data = [
      {
        label: "Menu One",
        onPress: jest.fn(),
      },
      {
        label: "Menu Two",
        onPress: jest.fn(),
      },
    ];
    const testRenderer = renderer.create(<ListPopUp data={data} />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
