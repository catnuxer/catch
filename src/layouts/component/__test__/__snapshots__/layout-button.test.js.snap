// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout-button ListPopUp test snapshoot listpopup 1`] = `
.emotion-3 {
  position: absolute;
  background: white;
  width: 100%;
  top: 100px;
  z-index: 2;
  box-shadow: rgba(0,0,0,0.1) 0px 1px 6px 0px;
}

.emotion-2 {
  list-style-type: none;
  margin: 0px;
  padding: 0px;
}

.emotion-0 {
  font-size: 14px;
  margin: 0px;
  padding: 15px;
}

.emotion-0:hover {
  background-color: #a9ddf9;
  cursor: pointer;
}

<div
  className="emotion-3"
>
  <ul
    className="emotion-2"
  >
    <li
      className="emotion-0"
      onClick={[MockFunction]}
    >
      <span>
        Menu One
      </span>
    </li>
    <li
      className="emotion-0"
      onClick={[MockFunction]}
    >
      <span>
        Menu Two
      </span>
    </li>
  </ul>
</div>
`;

exports[`layout-button test snappshoot button 1`] = `
.emotion-0 {
  width: 100%;
  padding: 15px;
  border: 1px solid #e0e0e0;
  border-radius: 5px;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-0:active {
  border: 1px solid #005a8b;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
>
  test
</button>
`;
